### PR TITLE
feat!: bump major version to v2

### PR DESCRIPTION
**Pull request by kurtisvg**
## Change Description

First step towards #872

### CODE REVIEW

File: proxy/certs/certs.go [hunk 1]
```diff
@@ -228,7 +228,7 @@ func (s *RemoteCertSource) Local(instance string) (tls.Certificate, error) {
 			return tls.Certificate{}, tokErr
 		}
 		// TODO: remove this once issue with OAuth2 Tokens is resolved.
-		// See https://github.com/GoogleCloudPlatform/cloudsql-proxy/issues/852.
+		// See https://github.com/GoogleCloudPlatform/cloudsql-proxy/v2/issues/852.
```
Comment (enocom) [inline proxy/certs/certs.go:231]: False positive.

File: proxy/proxy/client.go [hunk 2]
```diff
@@ -219,7 +219,7 @@ func (c *Client) refreshCfg(instance string) (addr string, cfg *tls.Config, vers
 		Certificates: []tls.Certificate{mycert},
 		RootCAs:      certs,
 		// We need to set InsecureSkipVerify to true due to
-		// https://github.com/GoogleCloudPlatform/cloudsql-proxy/issues/194
+		// https://github.com/GoogleCloudPlatform/cloudsql-proxy/v2/issues/194
```
Comment (enocom) [inline proxy/proxy/client.go:222]: False positive.

File: .github/workflows/tests.yaml [hunk 3]
```diff
@@ -20,20 +20,20 @@ jobs:
...
-    - run: goimports -w .
-    - run: go mod tidy
-    - name: Verify no changes from goimports and go mod tidy. If you're reading this and the check has failed, run `goimports -w . && go mod tidy`.
-      run: git diff --exit-code 
-    - name: check for mismatch
-      shell: bash
-      run: |
-        VERSION=$(cat version.txt)
-        grep -w \"$VERSION\" cmd/cloud_sql_proxy/cloud_sql_proxy.go || exit 1
+      - name: Setup Go
```
Comment (enocom) [inline .github/workflows/tests.yaml:23]: nit: whitespace

File: go.mod [hunk 4]
```diff
@@ -1,6 +1,6 @@
-module github.com/GoogleCloudPlatform/cloudsql-proxy
+module github.com/GoogleCloudPlatform/cloudsql-proxy/v2
 
-go 1.13
+go 1.17
```
Comment (enocom) [inline go.mod:3]: What's the rationale for this change?

**Review by enocom — APPROVED** [review]
Minor issue in go.mod and some nits.

File: .github/workflows/tests.yaml (hunk 3 above)
Comment (kurtisvg) [reply .github/workflows/tests.yaml:23]: Which whitespace? The indent is how the yaml formatter seems to think it should be

Comment (kurtisvg) [reply .github/workflows/tests.yaml:23]: Which whitespace? The indent is how the yaml formatter seems to think it should be

Comment (enocom) [reply .github/workflows/tests.yaml:23]: OK. I see the VS Code formatter does this.

**Review by enocom — APPROVED** [review]
